nomad/structs/acl.go

```diff
@@ -1121,28 +1121,36 @@ func (a *ACLAuthMethodConfig) Canonicalize() {
 		if len(a.OIDCClientAssertion.Audience) == 0 {
 			a.OIDCClientAssertion.Audience = []string{a.OIDCDiscoveryURL}
 		}
+		// move the client secret into the client assertion
 		a.OIDCClientAssertion.ClientSecret = a.OIDCClientSecret
+		// do not also send the client secret normally
+		a.OIDCClientSecret = ""
 		a.OIDCClientAssertion.Canonicalize()
 	}
 }
 
-func (a *ACLAuthMethodConfig) Validate() error {
+func (a *ACLAuthMethodConfig) Validate(methodType string) error {
 	if a == nil {
 		return errors.New("missing auth method Config")
 	}
 	mErr := &multierror.Error{}
-	if a.OIDCDiscoveryURL == "" {
-		mErr = multierror.Append(mErr, errors.New("missing OIDCDiscoveryURL"))
-	}
-	if a.OIDCClientID == "" {
-		mErr = multierror.Append(mErr, errors.New("missing OIDCClientID"))
-	}
-	if len(a.BoundAudiences) == 0 || a.BoundAudiences[0] == "" {
-		mErr = multierror.Append(mErr, errors.New("missing BoundAudiences"))
-	}
```

Comment:
This doesn't appear to be validated anywhere now and we removed the test assertion. Are we deprecating this field? Is it just always ignored for OIDC?

Reply:
Ok, that feels like the right move. We're trying our best to validate this stuff but there's a certain degree to which "what is needed is provider-defined" we're not going to be able to work around.